config/peerpods/podvm/aws-VM-image-delete-job.yaml

```diff
@@ -41,8 +41,8 @@ spec:
           export MAC=$(curl -m 3 -s http://169.254.169.254/latest/meta-data/mac)
           [[ ! "${AWS_REGION}" ]] && export AWS_REGION=$(curl -m 30 -s --show-error http://169.254.169.254/latest/meta-data/placement/region)
           [[ ! "${AWS_REGION}" ]] && echo "AWS_REGION is missing" && exit 1
-          export IMAGE_NAME=${IMAGE_NAME:-peer-pod-ami}
-          PODVM_AMI_ID=$(aws ec2 describe-images --query "Images[*].[ImageId]" --filters "Name=name,Values=${IMAGE_NAME}" --region ${AWS_REGION} --output text) && \
           echo "Deleting AMI: ${PODVM_AMI_ID}"
-          aws ec2 deregister-image --image-id ${PODVM_AMI_ID} --region ${AWS_REGION}
-          echo "Deleting AMI: ${PODVM_AMI_ID} - DONE"
+          RES=$(aws ec2 deregister-image --image-id "${PODVM_AMI_ID}" --region "${AWS_REGION}" 2>&1) || ERR=true
+          echo ${RES}
+          [[ ${ERR} ]] && [[ "$RES" =~ InvalidAMIID\.(Unavailable|NotFound) ]] # if deregister returned error and image is already deleted, continue
+          echo "Deleted AMI: ${PODVM_AMI_ID} - DONE"
```

Comment:
Are users supposed to see the error message in the output of the job?  Is that captured in a log anywhere?  Also a little weird that we say the AMI is deleted when it didn't exist.  I feel like there is an "if" statement missing.

Reply:
It's possible to see the error in the job logs, however, in this case the error means image is already deleted and there's nothing else to do which is fine, would it make more sense to change the wording to something like: "image deletion job completed"?

Reply:
> Are users supposed to see the error message in the output of the job?

Users can see these logs with `oc logs` but they aren't supposed to look at them. This is an internal job.

> Is that captured in a log anywhere? 

This should probably be captured by the OSC must-gather if it isn't the case already.

> It's possible to see the error in the job logs, however, in this case the error means image is already deleted and there's nothing else to do which is fine, would it make more sense to change the wording to something like: "image deletion job completed"?

IIUC the only case where we get the `InvalidAMIID\.(Unavailable|NotFound)` error is when someone deleted the image automatically created by us in our back, right ? Assuming this is `fine`, then the job just needs to be idempotent and report something like `"PODVM_AMI_ID: \"$PODVM_AMI_ID\" is deleted"`.